Array.cc

```diff
@@ -528,6 +538,14 @@ void Array::append_dim(int size, const string &name)
     update_length();
 }
 
+void Array::append_dim_ll(int64_t size, const string &name)
+{
+    dimension d(size, www2id(name));
+    _shape.push_back(d);
```

Comment:
Here's a place where emplace_back() can be used. You can write '_shape.emplace_back(size, www2id(name));' and it will avoid the creation of a temporary object and a call to the copy constructor.

Reply:
accepted both suggestions and changed the code accordingly.